units/src/amount/mod.rs

```diff
@@ -91,6 +91,9 @@ pub enum Denomination {
     Bit,
     /// satoshi (1 BTC = 100,000,000 satoshi).
     Satoshi,
+    /// Stops users from casting this enum to an integer.
+    #[doc(hidden)]
+    _Never(Infallible),
```

Comment:
The previous name was better, just the underscore was something worth adding.

Reply:
I thought that because it may show up in rust-analyzer that a really long name was annoying and `_Never` would be less so?

Reply:
Why would it be annoying? But anyway, it only shows up in our crate, not downstream. If you want something terse that conveys the intent name it `_DoNotUse` (and explicitly tell in comment that this may get removed if one day Rust supports disabling casts natively).

Reply:
Used as suggested.